src/viking.js

```diff
@@ -1,11 +1,82 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health = health;
+    this.strength = strength;
+}
+
+Soldier.prototype.attack = function(){
+    return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function(damage){
+    this.health -= damage;
+}
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = name;
+}
+
+Viking.prototype = new Soldier();
+// Viking.prototype = Object.create(Solider.prototype); //for some reason this didn't work
+Viking.prototype.constructor = Viking;
+Viking.prototype.receiveDamage = function(damage){
+    this.health -= damage;
+    return this.health>0? `${this.name} has received ${damage} points of damage`:`${this.name} has died in act of combat`;
+}
+
+Viking.prototype.battleCry = function(){
+    return "Odin Owns You All!";
+}
+
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+    Soldier.call(this, health, strength);
+
+    this.receiveDamage = function(damage){
+        this.health -= damage;
+        return this.health>0? `A Saxon has received ${damage} points of damage`:`A Saxon has died in combat`;
+    };
+}
+
+Saxon.prototype = new Soldier();
+Saxon.prototype.constructor = Saxon;
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy =[];
+    this.saxonArmy = [];
+    this.addViking = function(Viking){
+        this.vikingArmy.push(Viking);
+    };
+    this.addSaxon = function(Saxon){
+        this.saxonArmy.push(Saxon);
+    };
+    this.vikingAttack = function(){
+        let index = Math.floor(Math.random() * this.saxonArmy.length);
```

Comment:
It works ok but ideally it would be better to calculate a random index for each kind of soldier (the vikingArmy and saxonArmy arrays can have different lengths)